99_helpers/create_sample_data.R

```diff
@@ -0,0 +1,355 @@
+      "MSNA_data.csv"
+    )) %>%
+      mutate(
+        across(
+          .cols = protection_lsg:livelihoods_lsg,
+          .fns = ~ case_when(
+            . == "4+" ~ 5,
+            is.na(.) ~ NA_real_,
+            TRUE ~ as.numeric(.)
+          )
+        ),
+        across(
+          .cols = matches("admin[1-3]"),
+          .fns = ~ gsub(
+            "[ ]|[-]|[.]|[_]|[,]",
+            "",
+            tolower(stringi::stri_trans_general(.x, "latin-ascii"))
+          )
+        )
+      ) %>%
+      pivot_longer(
+        cols = ends_with("_lsg"),
+        names_to = "sector",
+        values_to = "severity"
+      ) %>%
+      filter(
+        !is.na(severity),
+        admin0 %in% df$adm0_pcode
+      ) %>%
+      transmute(
+        uuid,
+        adm0_pcode = admin0,
+        adm_name = case_when(
+          adm0_pcode == "COL" ~ admin1,
+          !is.na(admin3_hno) ~ admin3_hno,
+          admin2_hno == "baidoa" ~ "baydhaba",
+          !is.na(admin2_hno) ~ admin2_hno,
+          !is.na(admin1) ~ admin1
+        ),
+        adm_pcode = df$adm_pcode[match(adm_name, df$adm_name)],
+        sector = case_when(
+          sector == "edu_lsg" ~ "Education",
+          sector == "foodsec_lsg" ~ "FS/FSL",
+          sector == "health_lsg" ~ "Health",
+          sector == "markets_er_liv_lsg" ~ "ERL",
+          sector == "protection_lsg" ~ "Protection",
+          sector == "shelter_lsg" ~ "Shelter",
+          sector == "wash_lsg" ~ "WASH"
+        ),
+        severity,
+        weight = as.numeric(weights)
+      ) %>%
+      filter(!is.na(adm_pcode)) %>%
+      select(-adm_name)
+
+    df_sectors <- df
+
+    if (gender_age_disagg) {
+      df_sectors <- df_sectors %>%
+        filter(!is.na(sex) & !is.na(age))
+    }
+
+    if (pop_group_disagg) {
+      df_sectors <- df_sectors %>%
+        filter(!is.na(population_group))
+    }
+
+    temp <- df_sectors %>%
+      mutate(
+        disaggregation = case_when(
+          gender_age_disagg & pop_group_disagg ~
+            paste(
+              population_group,
+              sex,
+              age
+            ),
+          pop_group_disagg ~ population_group,
+          gender_age_disagg ~ paste(sex, age),
+          TRUE ~ NA_character_
+        )
+      ) %>%
+      filter(
+        adm0_pcode %in% unique(df_msna$adm0_pcode),
+        adm_pcode %in% df_msna$adm_pcode
+      ) %>%
+      group_by(
+        adm0_pcode,
+        adm_pcode,
+        disaggregation,
+        sector
+      ) %>%
+      summarize(
+        pin = sum(round(pin), na.rm = TRUE),
+        affected_population = sum(round(affected_population), na.rm = TRUE),
+        .groups = "drop_last"
+      ) %>%
+      mutate(affected_population = max(affected_population, na.rm = TRUE)) %>%
+      ungroup() %>%
+      arrange(
+        adm0_pcode,
+        adm_pcode,
+        disaggregation,
+        sector
+      )
+
+    df_data <- temp %>%
+      filter(adm0_pcode == sample(unique(temp$adm0_pcode), 1)) %>%
```

Comment:
It's just randomly selecting a single country then here, right? Can you add a seed in the function call that allows us to repeat the same randomisation. Are there any other random parts?